jobs/keepalived/templates/bin/post-start.erb

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+	
+set -e # exit immediately if a simple command exits with a non-zero status
+set -u # report the usage of uninitialized variables
+# source properties file here.
+export IS_HA=<%= link("broker").p("ha_enabled") %>
+iaas_name='<%= p('iaas_config.name', nil) %>'
+
+if [[ "$iaas_name" == "gcp" && "$IS_HA" == "true" ]]; then
+  echo "gcp post-start script execution has started"	
+	
+  export AUTHBASEURL="-authbaseurl=https://accounts.google.com/o/oauth2/token"
+  export PRIVATEKEYID="-privatekeyid=<%= p("iaas_config.credentials.private_key_id") %>"
+  export CLIENTEMAILID="-clientemailid=<%= p("iaas_config.credentials.client_email") %>"
+  export GCPBASEURL="-gcpbaseurl=https://www.googleapis.com"
+  export PROJECTID="-projectid=<%= p("iaas_config.credentials.project_id") %>"
+  export DEPLOYMENTGUID="-deploymentguid=<%= spec.deployment  %>"
+  export FLOATINGIP="-floatingip=<%= link("broker").p("internal.ip") %>"
+  export INSTANCEIP="-instanceip=<%= spec.ip %>"
+  export SFBROKERPORT="-sfbrokerport=<%= link("broker").p("internal.port") %>"
+  export SFREPORTPORT="-sfreportport=<%= link("report").p("report.port") %>"
+  export SFEXTERNALPORT="-sfexternalport=<%= link("broker").p("external.port") %>"
+  export SFDEPHOOKSPORT="-sfdephooksport=<%= link("deployment_hooks").p("hook.port") %>"
+  export PROBEINTERVAL="-probeinterval=2"
+  export PROBEHEALTHCHECK="-probehealthcheck=2"
+  export PROBEPROTOCOL="-probeprotocol=Http"
+  export PROBEPORT="-probeport=9595"
+  export PROBEREQUESTPATH="-proberequestpath=/heartbeat"
+  export LANDSCAPE="-landscape=gcp"
+  export PRIVATEKEY="-privatekey=<%= p("iaas_config.credentials.private_key").gsub(/\r?\n/, "\\n") %>"
+  export SCOPES="--scopes=https://www.googleapis.com/auth/cloud-platform,https://www.googleapis.com/auth/compute"
+
+  export NETWORK="-network=<%= spec.networks.sf_service_fabrik_broker.cloud_properties.network_name %>"
+  export SUBNET="-subnetwork=<%= spec.networks.sf_service_fabrik_broker.cloud_properties.subnetwork_name %>"
+
+  exec /var/vcap/packages/ha-utils/bin/ha-utils ${AUTHBASEURL} ${PRIVATEKEYID} ${CLIENTEMAILID} ${GCPBASEURL} ${PROJECTID} ${NETWORK} ${SUBNET} ${DEPLOYMENTGUID} ${FLOATINGIP} ${INSTANCEIP} ${SFBROKERPORT} ${SFREPORTPORT} ${SFEXTERNALPORT} ${SFDEPHOOKSPORT} ${PROBEINTERVAL} ${PROBEHEALTHCHECK} ${PROBEPROTOCOL} ${PROBEPORT} ${PROBEREQUESTPATH} ${LANDSCAPE} "${PRIVATEKEY}" ${SCOPES}
```

Comment:
Too many command line args. cant we pass minimal args in cmd line and pass a config file? all the props set above can anyways be found via bosh links.

Reply:
where is stdout & stderr of hautils directed? I dont see any redirection here?

Reply:
it runs as part of post-start. So, the output is re-directed to the stdout and stderr of post-start of keepalived.

Reply:
Regarding the command line arguments.. Agreed! However, as discussed with you, parking this change for now, since it will involve a huge change/testing effort. Will further raise a bug and take it up.